Update for rust-libcryptsetup-rs 0.8

Clearly something has changed for 0.8 and unlike the 0.7 bump it'll need some updates for 0.8 :-/

````
error[E0277]: the trait bound `CryptParamsLuks2: libcryptsetup_rs::format::CryptParams` is not satisfied
   --> client-linuxapp/src/reencrypt/rebind.rs:157:59
    |
157 |         .load::<libcryptsetup_rs::CryptParamsLuks2>(None, None)
    |          ---- required by a bound introduced by this call ^^^^ the trait `libcryptsetup_rs::format::CryptParams` is not implemented for `CryptParamsLuks2`
    |
    = help: the following other types implement trait `libcryptsetup_rs::format::CryptParams`:
              ()
              CryptParamsIntegrityRef<'a>
              CryptParamsLoopaesRef<'a>
              CryptParamsLuks1Ref<'a>
              CryptParamsLuks2Ref<'a>
              CryptParamsPlainRef<'a>
              CryptParamsTcryptRef<'a>
              CryptParamsVerityRef<'a>
note: required by a bound in `CryptContextHandle::<'a>::load`
   --> /usr/share/cargo/registry/libcryptsetup-rs-0.8.0/src/context.rs:130:20
    |
130 |     pub fn load<T: CryptParams>(
    |                    ^^^^^^^^^^^ required by this bound in `CryptContextHandle::<'a>::load`
error[E0277]: the trait bound `CryptParamsLuks2: libcryptsetup_rs::format::CryptParams` is not satisfied
   --> client-linuxapp/src/reencrypt/mod.rs:67:59
    |
67  |         .load::<libcryptsetup_rs::CryptParamsLuks2>(None, None)
    |          ---- required by a bound introduced by this call ^^^^ the trait `libcryptsetup_rs::format::CryptParams` is not implemented for `CryptParamsLuks2`
    |
    = help: the following other types implement trait `libcryptsetup_rs::format::CryptParams`:
              ()
              CryptParamsIntegrityRef<'a>
              CryptParamsLoopaesRef<'a>
              CryptParamsLuks1Ref<'a>
              CryptParamsLuks2Ref<'a>
              CryptParamsPlainRef<'a>
              CryptParamsTcryptRef<'a>
              CryptParamsVerityRef<'a>
note: required by a bound in `CryptContextHandle::<'a>::load`
   --> /usr/share/cargo/registry/libcryptsetup-rs-0.8.0/src/context.rs:130:20
    |
130 |     pub fn load<T: CryptParams>(
    |                    ^^^^^^^^^^^ required by this bound in `CryptContextHandle::<'a>::load`
error[E0277]: the trait bound `CryptParamsLuks2: libcryptsetup_rs::format::CryptParams` is not satisfied
   --> client-linuxapp/src/serviceinfo.rs:256:63
    |
256 |             .load::<libcryptsetup_rs::CryptParamsLuks2>(None, None)
    |              ---- required by a bound introduced by this call ^^^^ the trait `libcryptsetup_rs::format::CryptParams` is not implemented for `CryptParamsLuks2`
    |
    = help: the following other types implement trait `libcryptsetup_rs::format::CryptParams`:
              ()
              CryptParamsIntegrityRef<'a>
              CryptParamsLoopaesRef<'a>
              CryptParamsLuks1Ref<'a>
              CryptParamsLuks2Ref<'a>
              CryptParamsPlainRef<'a>
              CryptParamsTcryptRef<'a>
              CryptParamsVerityRef<'a>
note: required by a bound in `CryptContextHandle::<'a>::load`
   --> /usr/share/cargo/registry/libcryptsetup-rs-0.8.0/src/context.rs:130:20
    |
130 |     pub fn load<T: CryptParams>(
    |                    ^^^^^^^^^^^ required by this bound in `CryptContextHandle::<'a>::load`
For more information about this error, try `rustc --explain E0277`.
error: could not compile `fdo-client-linuxapp` (bin "fdo-client-linuxapp") due to 3 previous errors
````

--- a/client-linuxapp/Cargo.toml
+++ b/client-linuxapp/Cargo.toml
@@ -17,7 +17,7 @@ rand = "0.8.4"
 nix = "0.26"
 uuid = "1.3"
 thiserror = "1"
-libcryptsetup-rs = { version = "0.7.0", features = ["mutex"] }
+libcryptsetup-rs = { version = "0.8.0", features = ["mutex"] }
 secrecy = "0.8"
 devicemapper = "0.33"
 

--- a/client-linuxapp/src/reencrypt/mod.rs
+++ b/client-linuxapp/src/reencrypt/mod.rs
@@ -64,7 +64,7 @@ fn perform_reencrypt(dev_name: &str) -> Result<()> {
         .context("Error opening device")?;
 
     dev.context_handle()
-        .load::<libcryptsetup_rs::CryptParamsLuks2>(None, None)
+        .load::<()>(None, None)
         .context("Error loading device context")?;
 
     let status = dev

--- a/client-linuxapp/src/reencrypt/rebind.rs
+++ b/client-linuxapp/src/reencrypt/rebind.rs
@@ -154,7 +154,7 @@ fn clevis_bind(
     log::trace!("Clevis bind successful");
 
     dev.context_handle()
-        .load::<libcryptsetup_rs::CryptParamsLuks2>(None, None)
+        .load::<()>(None, None)
         .context("Error re-loading device context")?;
 
     log::trace!("Reloaded device context");

--- a/client-linuxapp/src/serviceinfo.rs
+++ b/client-linuxapp/src/serviceinfo.rs
@@ -253,7 +253,7 @@ impl DiskEncryptionInProgress {
         log::debug!("Device initiated");
 
         dev.context_handle()
-            .load::<libcryptsetup_rs::CryptParamsLuks2>(None, None)
+            .load::<()>(None, None)
             .context("Error loading device context")?;
 
         log::debug!("Device information loaded");

